changelog.d/8367.misc

```diff
@@ -0,0 +1 @@
+Fixes the way synapse distinguishes between runtime and developer dependencies.
```

Comment:
We should use the same newsfragment as the PR that regressed this so that they'll show up together in the changelog.